src/modeler/AutoRest.Swagger.Tests/Resource/Swagger/Validation/required-readonly-properties-3.json

```diff
@@ -0,0 +1,114 @@
+﻿{
+  "swagger": "2.0",
+  "info": {
+    "title": "Uses 'required' & 'readonly' properties.",
+    "description": "Show an error/warning if a definition property is marked readOnly: true plus it appears under the required array of properties.",
+    "version": "2014-04-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "basePath": "/",
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "paths": {
+    "/foo": {
+      "get": {
+        "tags": [ "SampleTag" ],
+        "operationId": "Foo_Get",
+        "description": "Test Description",
+        "parameters": [
+          {
+            "name": "parameter_1",
+            "in": "path",
+            "required": true,
+            "description": "Test Description",
+            "schema": {
+              "properties": {
+                "id": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "Resource Id"
+                },
+                "name": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "Resource name"
+                },
+                "type": {
+                  "readOnly": true,
+                  "type": "string",
+                  "description": "Resource type"
+                },
+                "location": {
+                  "type": "string",
+                  "description": "Resource location"
+                },
+                "tags": {
+                  "type": "object",
+                  "additionalProperties": {
+                    "type": "string"
+                  },
+                  "description": "Resource tags"
+                }
+              },
+              "required": [
+                "location", "name"
+              ],
+              "description": "The Resource definition."
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ResponseResource"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ResponseResource": {
```

Comment:
nit: moving forward, let's create a json with common definitions for resource types, our tests should reflect the ideal world we wish to see